sql: TestRelocateNonVoters failed [trying to add NON_VOTER to a store that already has a LEARNER]

sql.TestRelocateNonVoters [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20994170?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20994170?buildTab=artifacts#/) on release-26.1 @ [0fc80f178e797c71cd2e78958a27974352f65bae](https://github.com/cockroachdb/cockroach/commits/0fc80f178e797c71cd2e78958a27974352f65bae):


```
=== RUN   TestRelocateNonVoters
    test_log_scope.go:171: test logs captured to: /artifacts/tmp/_tmp/f4d1e5a362e4384d18c9c26e2defb957/logTestRelocateNonVoters1600281478
    test_log_scope.go:82: use -show-logs to present logs inline
    multitenant_admin_function_test.go:966: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/f4d1e5a362e4384d18c9c26e2defb957/logTestRelocateNonVoters1600281478
--- FAIL: TestRelocateNonVoters (69.15s)
=== RUN   TestRelocateNonVoters/ALTER_RANGE_x_RELOCATE_NONVOTERS
    multitenant_admin_function_test.go:383: condition failed to evaluate within 45s: from multitenant_admin_function_test.go:420: expected "ok" to be contained in result "trying to add({ChangeType:ADD_NON_VOTER Target:n1,s1}) to a store that already has a LEARNER" tenant=secondary query=``ALTER RANGE (SELECT min(range_id) FROM [SHOW RANGES FROM TABLE t]) RELOCATE NONVOTERS FROM 5 TO 1;`` leaseholder=4 replicas=[2 3 4 5] voting_replicas=[2 3 4] non_voting_replicas=[5] fromReplica=5 toReplica=1 row=0 col=2
--- FAIL: TestRelocateNonVoters/ALTER_RANGE_x_RELOCATE_NONVOTERS (59.06s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRelocateNonVoters.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-58824

--- a/pkg/sql/multitenant_admin_function_test.go
+++ b/pkg/sql/multitenant_admin_function_test.go
@@ -934,6 +934,27 @@ func TestRelocateNonVoters(t *testing.T) {
 					require.NoErrorf(t, err, message)
 					err = testCluster.WaitForFullReplication()
 					require.NoErrorf(t, err, message)
+					testutils.SucceedsSoon(t, func() error {
+						// Check for learners before disabling the replicate queues to
+						// allow cleanup of learners
+						rows, err := db.QueryContext(ctx,
+							`SELECT learner_replicas FROM [SHOW RANGES FROM INDEX t@primary WITH DETAILS]`)
+						if err != nil {
+							return err
+						}
+						defer rows.Close()
+						if rows.Next() {
+							var learners string
+							err = rows.Scan(&learners)
+							if err != nil {
+								return err
+							}
+							if learners != "{}" && learners != "" {
+								return errors.Newf("waiting for LEARNER replicas: %s", learners)
+							}
+						}
+						return nil
+					})
 					testCluster.ToggleLeaseQueues(false)
 					testCluster.ToggleReplicateQueues(false)
 					testCluster.ToggleSplitQueues(false)

